### PR TITLE
Feat filter show

### PR DESCRIPTION
**Pull request by anitavincent**
#### What is the purpose of this pull request?
Add three new components to enable users to add hidden filters that are managed by another filter

#### How should this be manually tested?
check filters/advance story

#### Screenshots or example usage
image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/filters/filter-toggler/filter-visibility-state.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  const onChange = () => {
+    // action performed before new value for appliedItems is saved
+    const oldItems = appliedItems
+    const firstNewSelectedItem = items.find((item) => {
+      const isNew = !oldItems.find((i) => i.id === item.id)
+
+      return isSelected(item) && isNew
+    })
+
+    setFilterOpenOnMount(firstNewSelectedItem)
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-toggler/filter-visibility-state.tsx:40]: I think you can do it without using a state, for example:

`const filterOpenOnMount = useMemo(() => items.find((item) => {
      const isNew = !appliedItems.find((i) => i.id === item.id)

      return isSelected(item) && isNew
    }), [items])
`
this way you'll avoid one re-render

Comment (lucasaarcoverde) [reply packages/admin-ui/src/filters/filter-toggler/filter-visibility-state.tsx:40]: You can also improve the algorithm to check the filter to be opened on mount, the current solution takes O(n^2). It's ok to open in other PR

File: packages/admin-ui/src/filters/filter-toggler/filter-visibility-state.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    state.onChange()
+  }
+
+  return {
+    addFilter,
+    removeFilter,
+    items,
+    filterState: { ...state, onChange },
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-toggler/filter-visibility-state.tsx:49]: Does this state control the FilterVisibility component or the optional filters rendered?

Comment (anitavincent) [reply packages/admin-ui/src/filters/filter-toggler/filter-visibility-state.tsx:49]: It controls `FilterToggleVisible`. That includes the "show more" button and popover

File: packages/admin-ui/src/filters/__stories__/filter-advanced.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    filterStates: [],
+  })
+
+  return (
+    <FilterGroup state={filterGroupState}>
+      <GenericFilter list={list1} label="Important filter" />
+      <GenericFilter list={list2} label="Price" />
+      <GenericFilter list={list1} label="Status" />
+
+      <FilterOptional id="#rare" label="Rare" state={togState}>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/__stories__/filter-advanced.stories.tsx:72]: It occurred to me that the label is not being used here, is it still necessary?

**Review by lucasaarcoverde — APPROVED** [review]
LGTM, just a question

File: packages/admin-ui/src/filters/__stories__/filter-advanced.stories.tsx (hunk 3 above)
Comment (anitavincent) [reply packages/admin-ui/src/filters/__stories__/filter-advanced.stories.tsx:72]: We are, it's used on the filtercontrol list

Comment (lucasaarcoverde) [reply packages/admin-ui/src/filters/__stories__/filter-advanced.stories.tsx:72]: That's true! Sorry

**Pull request merged**